data/types.go

```diff
@@ -36,17 +38,24 @@ type Signature struct {
 	Signature HexBytes `json:"sig"`
 }
 
-type Key struct {
-	Type       string   `json:"keytype"`
-	Scheme     string   `json:"scheme"`
-	Algorithms []string `json:"keyid_hash_algorithms,omitempty"`
-	Value      KeyValue `json:"keyval"`
+type PublicKey struct {
+	Type       string          `json:"keytype"`
+	Scheme     string          `json:"scheme"`
+	Algorithms []string        `json:"keyid_hash_algorithms,omitempty"`
+	Value      json.RawMessage `json:"keyval"`
 
 	ids    []string
 	idOnce sync.Once
 }
 
-func (k *Key) IDs() []string {
+type PrivateKey struct {
+	Type       string          `json:"keytype"`
+	Scheme     string          `json:"scheme,omitempty"`
+	Algorithms []string        `json:"keyid_hash_algorithms,omitempty"`
+	Value      json.RawMessage `json:"keyval"`
+}
+
+func (k *PublicKey) IDs() []string {
 	k.idOnce.Do(func() {
 		data, _ := cjson.Marshal(k)
```

Comment:
Is it safe to ignore this error?

Reply:
You'll never hit this error from deserializing a bad key in metadata since that'll fail already on ingestion when unmarshalling -- so you can assume that `PublicKey` is valid unless somehow a key implementation purposely puts in some invalid data when they implement `PublicData` https://github.com/theupdateframework/go-tuf/blob/736fcc61962c9cece79df91d0e8393e351d0fd8f/pkg/keys/ed25519.go#L120

Testing this is weird, because if it does fail, it'll still produce a hash, just a bad hash -- what do you think about a `panic`?

Reply:
panic is good. Something should really be wrong if this fails!